pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/SegmentAssignmentUtils.java

```diff
@@ -86,6 +86,55 @@ private SegmentAssignmentUtils() {
     return instances;
   }
 
+  /**
+   * Assigns the segment for the non-replica-group based segment assignment strategy and returns the assigned instances.
+   */
+  static List<String> assignSegmentWithoutReplicaGroup(Map<String, Map<String, String>> currentAssignment,
+      InstancePartitions instancePartitions, int replication) {
+    List<String> instances =
+        SegmentAssignmentUtils.getInstancesForNonReplicaGroupBasedAssignment(instancePartitions, replication);
+    int[] numSegmentsAssignedPerInstance = getNumSegmentsAssignedPerInstance(currentAssignment, instances);
+    int numInstances = numSegmentsAssignedPerInstance.length;
+    PriorityQueue<Pairs.IntPair> heap = new PriorityQueue<>(numInstances, Pairs.intPairComparator());
```

Comment:
I assume `Pairs.intPairComparator` that this will be an ascending order?

Reply:
Yes, it is sorted on the left value firstly, then right value secondly.